zerver/models.py

```diff
@@ -2646,9 +2646,11 @@ class CustomProfileField(models.Model):
         (DATE, str(_('Date picker')), check_date, str, "DATE"),
         (URL, str(_('Link')), check_url, str, "URL"),
         (EXTERNAL_ACCOUNT, str(_('External account')), check_short_string, str, "EXTERNAL_ACCOUNT"),
-    ]  # type: FieldTypeData
+    ]  # type: List[FieldElement]
 
-    ALL_FIELD_TYPES = FIELD_TYPE_DATA + CHOICE_FIELD_TYPE_DATA + USER_FIELD_TYPE_DATA
+    ALL_FIELD_TYPES = (cast(FieldTypeData, FIELD_TYPE_DATA) +
+                       cast(FieldTypeData, CHOICE_FIELD_TYPE_DATA) +
+                       cast(FieldTypeData, USER_FIELD_TYPE_DATA))
```

Comment:
We can avoid the unchecked casts here by spelling this `[*FIELD_TYPE_DATA, *CHOICE_FIELD_TYPE_DATA, *USER_FIELD_TYPE_DATA]`.

Reply:
Cool, updated. (Could you perhaps explain why this works?)

Reply:
I thought mypy was doing something where `[*[foo1, foo2], *[bar1, bar2]]` infers `List[Union[Foo, Bar]]`, but actually it just gives you `List[FooBarCommonBase]`.  In our case that works out to `List[Tuple[int, str, function, Callable[[Any], Any], str]]`.  So maybe we should be a bit more specific:

```python
ALL_FIELD_TYPES = [*FIELD_TYPE_DATA, *CHOICE_FIELD_TYPE_DATA, *USER_FIELD_TYPE_DATA]  # type: FieldTypeData
```